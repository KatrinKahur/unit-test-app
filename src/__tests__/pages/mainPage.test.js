import {render, screen} from "@testing-library/react";
import '@testing-library/jest-dom';
import user from '@testing-library/user-event';
import MainPage from "../../pages/mainPage";

describe('MainPage', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<MainPage />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('wrong file type triggers the error message', async () => {
        render(<MainPage/>);
        const fileContent = "some text"

        const file = new File([fileContent], 'test_file.java');
        File.prototype.text = jest.fn().mockResolvedValueOnce(fileContent);
        const input = screen.getByTestId("file-input");
        user.upload(input, file);

    })
})