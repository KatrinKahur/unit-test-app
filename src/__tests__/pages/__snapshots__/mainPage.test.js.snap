// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders correctly 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="toolName"
    >
      Java Unit Testing with AI (JUTAI)
    </h1>
    <form>
      <div
        class="inputButt"
      >
        <input
          data-testid="file-input"
          id="file-input"
          required=""
          type="file"
        />
      </div>
      <input
        class="testGeneratorButton"
        type="submit"
        value="Generate unit tests!"
      />
    </form>
    <div
      class="codeBlockWrapper"
    >
      <div
        class="requestBlockWrapper"
      >
        <h3>
          Java class: 
        </h3>
        <div
          class="requestBlock"
        >
          <pre
            data-testid="request-block"
            style="display: block; overflow-x: auto; padding: 0.5em; background: rgb(35, 35, 35); color: rgb(230, 225, 220);"
          >
            <code
              class="language-java"
              style="white-space: pre;"
            />
          </pre>
        </div>
      </div>
      <div
        class="responseBlockWrapper"
      >
        <h3>
          Tests: 
        </h3>
        <div
          class="responseBlock"
        >
          <pre
            style="display: block; overflow-x: auto; padding: 0.5em; background: rgb(35, 35, 35); color: rgb(230, 225, 220);"
          >
            <code
              class="language-java"
              style="white-space: pre;"
            />
          </pre>
        </div>
      </div>
      <form
        class="testForm"
        id="result-form"
      >
        <div>
          <label
            class="inputSquare"
          >
            <p
              class="testText"
            >
              Line Coverage
            </p>
            <textarea
              class="inputBox"
              name="name"
              required=""
              type="text"
            />
          </label>
          <label
            class="inputSquare"
          >
            <p
              class="testText"
            >
              Branch Coverage
            </p>
            <textarea
              class="inputBox"
              name="name"
              required=""
              type="text"
            />
          </label>
          <label
            class="inputSquare"
          >
            <p
              class="testText"
            >
              Passed Unit Tests
            </p>
            <textarea
              class="inputBox"
              name="name"
              required=""
              type="text"
            />
          </label>
        </div>
        <input
          class="testSubmitButt"
          type="submit"
          value="Submit"
        />
      </form>
    </div>
  </div>
</DocumentFragment>
`;
